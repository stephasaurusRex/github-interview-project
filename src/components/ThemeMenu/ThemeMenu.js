import get from "lodash.get";
import React from "react";
import styled from "styled-components";

import { themeWrapper } from "../../styles/ThemeProvider/ThemeProvider";
import Menu from "../Menu/Menu";

const Item = styled.div`
  min-width: 120px;
  color: ${(props) => (props.isSelected ? get(props, "theme.palette.primary") : "#333")};
  font-size: 1rem;
  cursor: pointer;

  :hover {
    color:
  }
`;

const ThemeMenu = ({ style, className, themes, setTheme, selectedTheme }) => (
  <Menu style={style} className={className}>
    {
      themes.map((theme, idx) =>
        <Item
          key={idx}
          onClick={() => setTheme(theme)}
          isSelected={selectedTheme.name === theme.name}>
          {theme.name}
        </Item>)
    }
  </Menu>);

ThemeMenu.defaultProps = {
  themes: [],
  setTheme: () => { },
  selectedTheme: {},
};

export default themeWrapper(ThemeMenu);
