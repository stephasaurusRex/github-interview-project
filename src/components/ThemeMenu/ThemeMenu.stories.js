import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import ThemeProvider from "../../styles/ThemeProvider/ThemeProvider";
import ThemeMenu from "./ThemeMenu";

storiesOf("ThemeMenu", module)

  .add("basic usage", withInfo(`Dropdown for changing the theme`)(
    () =>
      <ThemeProvider>
        <ThemeMenu />
      </ThemeProvider>));
