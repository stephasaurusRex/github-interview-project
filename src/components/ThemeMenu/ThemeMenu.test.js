import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import ThemeMenu from "./ThemeMenu";

describe("ThemeMenu", () => {

  it("renders without props", () => {
    const tree = renderer.create(<ThemeMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
