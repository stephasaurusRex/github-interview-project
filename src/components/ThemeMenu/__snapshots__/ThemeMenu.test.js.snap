// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeMenu renders without props 1`] = `
.c0 {
  z-index: 10;
  padding: 20px;
  background: transparent;
}

<div
  className="c0"
  style={undefined}
/>
`;
